include/nccl_ofi.h

```diff
@@ -194,6 +223,15 @@ typedef struct pending_reqs_q {
 	pending_reqs_q_elem_t *tail;
 } pending_reqs_q_t;
 
+typedef struct nccl_ofi_handle {
+	char ep_name[MAX_EP_ADDR];
+	uint64_t tag;
+	/* Save temporary communictor state when creating send communicator */
+	save_comm_state_t state;
```

Comment:
I'm not sure that this member makes sense here.  Because the size of the handle is severely limited (to `NCCL_NET_HANDLE_MAXSIZE`, or 64 bytes) I would argue that we should only keep information in the handle that can be interpreted by any rank.

What does this mean?  I think we should keep only the endpoint address and tag in the handle, and use a lookup table (a hashtable for example) to map the handle value to additional per-connection info.  The information there would be fetched from the remote ep by the initial tsend/trecv.

This would let us stop paying attention to the `NCCL_NET_HANDLE_MAXSIZE` limit, allow us to do proper versioning of the connection message, etc.

Reply:
If we happen to prepare a lookup table to map handle to additional connection information, how do you plan to share this hash table across remote peers? Using tsends/trecvs? How would you handle the case when you are not able to post tsend/trecv during connection establishment? Would you block on that?

Reply:
We already block on the connect message so it wouldn't be an extra burden, right?  

But we don't have to add this yet if it's not necessary.

Reply:
The change is to make `connect` and ``accept` as non-blocking. So, we can't do it this way.